src/fiatconnect/saga.ts

```diff
@@ -55,16 +65,74 @@ export function* handleFetchFiatConnectQuotes({
       flow,
       fiatConnectCashInEnabled,
       fiatConnectCashOutEnabled,
+      providerList: providerIds,
     })
     yield put(fetchFiatConnectQuotesCompleted({ quotes }))
   } catch (error) {
-    Logger.error(TAG, 'Could not parse dapps response', error)
     yield put(fetchFiatConnectQuotesFailed({ error: 'Could not fetch providers' }))
   }
 }
 
-export function* watchFetchFiatConnectQuotes() {
-  yield takeLeading(fetchFiatConnectQuotes.type, handleFetchFiatConnectQuotes)
+export function* handleFetchQuoteAndFiatAccount({
```

Comment:
I find this saga pretty complex. It seems like it's doing several things:
- checking that a user's fiat account ID is valid
- (if it's not, delete the fiat account)
- gets a quote
- checks that the quote is compatible with the given fiat account
- save the quote

To make it easier to reason about and debug, I'd suggest simplifying and making it more single-purpose. 

Ideas for a refactor:
- If you want to be defensive and do the fiat account ID check before attempting to create a transfer, you could try doing that before you even enter the ReviewFetchScreen component and take the fiat account ID as a prop (after all, you never actually want to enter that component unless there's a valid fiat account to use). 
- probably more effortful: migrate most/all of this to the backend. You're doing quite a lot of state management and API calls, which might be better organized into a single API such as `getMRUAccountQuote`, something like that. The difficulty could be checking whether the fiat account hasn't been deleted, since the user needs to be authenticated to call GET /accounts, but that seems like an edge case-- and one that could still be done separately in the client if we wanted.

Reply:
one unfortunate thing is that KYC is left out of this, which will likely add complexity. If anything that makes it more important to consider how to simplify now, though